tests/e2e/util.go

```diff
@@ -6200,3 +6200,146 @@ func getBlockDevSizeInBytes(f *framework.Framework, ns string, pod *v1.Pod, devi
 	output = strings.TrimSuffix(output, "\n")
 	return strconv.ParseInt(output, 10, 64)
 }
+
+// getVsanDPersistentVolumeClaimSpecWithStorageClass return the PersistentVolumeClaim
+// spec for vsanDirect datastore with specified storage class.
+func getVsanDPersistentVolumeClaimSpecWithStorageClass(namespace string, ds string,
+	storageclass *storagev1.StorageClass, pvcName string, podName string,
+	accessMode v1.PersistentVolumeAccessMode) *v1.PersistentVolumeClaim {
+	pvcAnnotations := make(map[string]string)
+	pvcAnnotations["volume.beta.kubernetes.io/storage-class"] = storageclass.Name
+	pvcAnnotations["placement.beta.vmware.com/storagepool_antiAffinityRequired"] = podName
+
+	pvclaimlabels := make(map[string]string)
+	pvclaimlabels["supervisor"] = "true"
+
+	disksize := diskSize
+	if ds != "" {
+		disksize = ds
+	}
+	if accessMode == "" {
+		// If accessMode is not specified, set the default accessMode.
+		accessMode = v1.ReadWriteOnce
+	}
+
+	claim := &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pvcName,
+			Namespace: namespace,
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			AccessModes: []v1.PersistentVolumeAccessMode{
+				accessMode,
+			},
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceName(v1.ResourceStorage): resource.MustParse(disksize),
+				},
+			},
+			StorageClassName: &(storageclass.Name),
+		},
+	}
+	claim.Labels = pvclaimlabels
+	claim.Annotations = pvcAnnotations
+	framework.Logf("pvc spec: %v", claim)
+	return claim
+}
+
+// getVsanDPodSpec returns pod spec for vsan direct datastore for a given persistentVolumeClaim
+func getVsanDPodSpec(ns string, nodeSelector map[string]string, pvclaims []*v1.PersistentVolumeClaim,
```

Comment:
you need to reset `generateName` in the spec

Reply:
I don't want generateName field in my spec as the name is hardcoded and with MakePod method generateName field is getting added, I have tried removing it but in vain. Please let me know a way if I can do that.